File: app/src/main/java/com/llbt/meepwn/lincolnblock/main/model/UserModel.java
package com.llbt.meepwn.lincolnblock.main.model;

import android.databinding.ObservableField;

import com.llbt.meepwn.lincolnblock.framework.base.Model;
import com.llbt.meepwn.lincolnblock.utils.binding.EditTextWatcher;
import com.llbt.meepwn.lincolnblock.utils.network.json_parser.JsonParser;

/**
 * package: com.llbt.meepwn.lincolnblock.main.model
 * author: MeePwn
 * email: maybewaityou@gmail.com
 * github: https://github.com/maybewaityou
 * date: 16/4/29 下午3:05
 * desc:
 */
public class UserModel extends Model {

	private ObservableField<String> lastName = new ObservableField<>("李四");
	private String name = "张三";
	private String imageUrl = "http://attach.bbs.miui.com/forum/201501/13/141755xbccajc97cbub1tb.png";
	private EditTextWatcher watcher = new EditTextWatcher() {
		@Override
		public void onTextChanged(CharSequence charSequence, int start, int before, int count) {
			if (lastName.get().equals(charSequence.toString())) {
				lastName.set(charSequence.toString());
			}
		}
	};

	public static UserModel initWithJsonString(String jsonString) {
		return JsonParser.parse(jsonString, UserModel.class);
	}

	public ObservableField<String> getLastName() {
		return lastName;
	}

	public void setLastName(ObservableField<String> lastName) {
		this.lastName = lastName;
	}

	public String getName() {
		return name;
	}

	public void setName(String name) {
		this.name = name;
	}

	public String getImageUrl() {
		return imageUrl;
	}

	public void setImageUrl(String imageUrl) {
		this.imageUrl = imageUrl;
	}

	public EditTextWatcher getWatcher() {
		return watcher;
	}

	public void setWatcher(EditTextWatcher watcher) {
		this.watcher = watcher;
	}

	@Override
	public String toString() {
		return "{" +
				"\"lastName\" : " + "\"" + lastName.get() + "\"" +
				", \"name\" : " + "\"" + name + "\"" +
				", \"imageUrl\" : " + "\"" + imageUrl + "\"" +
				", \"watcher\" : " + "\"" + watcher + "\"" +
				"}";
	}
}

File: app/src/main/java/com/llbt/meepwn/lincoinblock/main/model/UserModel.kt
package com.llbt.meepwn.lincoinblock.main.model

import android.databinding.ObservableField
import com.llbt.meepwn.lincoinblock.framework.base.BaseModel
import com.llbt.meepwn.lincoinblock.utils.binding.EditTextWatcher
import com.llbt.meepwn.lincoinblock.utils.network.json_parser.JsonParser

/**
 * package: com.llbt.meepwn.lincoinblock.main.model
 * author: MeePwn
 * email: maybewaityou@gmail.com
 * github: https://github.com/maybewaityou
 * date: 16/4/12 上午10:28
 * desc:
 */
class UserModel : BaseModel() {
    var lastName = ObservableField<String>("李四")
    var name: String? = "张三"
    var imageUrl = "http://attach.bbs.miui.com/forum/201501/13/141755xbccajc97cbub1tb.png"
    var watcher: EditTextWatcher = object : EditTextWatcher() {
        override fun onTextChanged(charSequence: CharSequence, i: Int, i1: Int, i2: Int) {
            if (lastName.get() != charSequence.toString()) {
                lastName.set(charSequence.toString())
            }
        }
    }

    companion object {
        fun initWithJsonString(jsonString: String): UserModel {
            return JsonParser.parse<UserModel>(jsonString, UserModel::class.java)
        }
    }

    override fun toString(): String {
        return "UserModel(lastName=${lastName.get()}, name=$name, imageUrl='$imageUrl', watcher=$watcher)"
    }

}
